modules/passkey/hardhat.config.ts

```diff
@@ -31,6 +31,18 @@ const customNetwork = CUSTOM_NODE_URL
     }
   : {}
 
+const compilerSettings = {
+  version: '0.8.24',
```

Comment:
if we're changing the compiler settings, what about using the latest version?

Reply:
My rationale for not doing it is as such:
1. from the changelog there are no significant features or fixes that were added since 0.8.24: [source](https://github.com/ethereum/solidity/blob/develop/Changelog.md)
2. Currently, the `FCL256Verifier` is byte-for-byte the same (and won't change with any of the planned changes :tada:, none of the audit competition changes affect it). Updating the compiler version would change that
3. The audit was done for compiler v0.8.24, I know we are changing the compiler settings, but it feels less extreme than a new compiler version.

I don't feel strongly either way, I did consider it when making this PR but opted to err on the side of caution.

Reply:
I ran the benchmarks with `0.8.26` and it seems it slightly improves runtime gas costs:
```
  Gas Benchmarking [@bench]
    SafeWebAuthnSignerProxy
      ⛽ deployment: 91042
      ✔ Benchmark signer deployment cost (568ms)
      ⛽ verification (FreshCryptoLib): 215607
      ✔ Benchmark signer verification cost with FreshCryptoLib verifier (107ms)
      ⛽ verification (Daimo): 345542
      ✔ Benchmark signer verification cost with Daimo verifier (100ms)
      ⛽ verification (Dummy): 12564
      ✔ Benchmark signer verification cost with Dummy verifier (107ms)
      ⛽ verification (Precompile): 13379
      ✔ Benchmark signer verification cost with Precompile verifier (117ms)
```

Reply:
With `0.8.25` they're almost identical except the Daimo verifier:
```
  Gas Benchmarking [@bench]
    SafeWebAuthnSignerProxy
      ⛽ deployment: 90399
      ✔ Benchmark signer deployment cost (508ms)
      ⛽ verification (FreshCryptoLib): 213844
      ✔ Benchmark signer verification cost with FreshCryptoLib verifier (145ms)
      ⛽ verification (Daimo): 342496
      ✔ Benchmark signer verification cost with Daimo verifier (121ms)
      ⛽ verification (Dummy): 12911
      ✔ Benchmark signer verification cost with Dummy verifier (110ms)
      ⛽ verification (Precompile): 13726
      ✔ Benchmark signer verification cost with Precompile verifier (111ms)
```

I guess those smile improvements (from the 0.8.26 changelog) made a slight difference, not a compiler engineer, though:
```
Compiler Features:
Yul IR Code Generation: Cheaper code for reverting with errors of a small static encoding size.
Yul Optimizer: New, faster default optimizer step sequence.
```

Reply:
Note that the code got slightly bigger from v0.8.25 -> v0.8.26

```
=== v0.8.25 ===
  Gas Benchmarking [@bench]
    SafeWebAuthnSignerProxy
      ⛽ deployment: 90399
      ✔ Benchmark signer deployment cost (508ms)

=== v0.8.26 ===
  Gas Benchmarking [@bench]
    SafeWebAuthnSignerProxy
      ⛽ deployment: 91042
      ✔ Benchmark signer deployment cost (568ms)
```

Overall, I think you are right, we can bump the compiler version here. My main argument was to reuse the FCLP256Verifier, but that is moot apparently :disappointed:

Reply:
> except the Daimo verifier:

This is slightly random because the verifiers have different gas characteristics depending on what public key/signature is used (which is random in the tests).

I would look specifically at `deployment`, `dummy` and `precompile` gas metrics only (unless you collect data from many runs and compute averages for FCL and Daimo).

Also, Daimo verifier is **not** compiled (we vendor bytecode directly), so compiler settings should not affect it at all.

Reply:
IMplemented suggestion in a6cd15a0c0c3e8b93a18e5ffa9975824c23c66ea